app/web_framework/cluster_master_application.py

```diff
@@ -64,9 +66,31 @@ def initialize(self, route_node=None, cluster_master=None):
         :type route_node: RouteNode | None
         :type cluster_master: ClusterMaster | None
         """
+        self._logger = log.get_logger(__name__)
         self._cluster_master = cluster_master
         super().initialize(route_node)
 
+    def prepare(self):
+        """
+        If the request has specified the session id, which is optional, and the session id does not match
+        the current instance's session id, then the client is asking for a resource that has expired and
+        no longer exists.
+        """
+        session_id = self.request.headers.get(SessionId.SESSION_HEADER_KEY)
+
+        if session_id is not None and session_id != SessionId.get():
+            raise ItemNotFoundError('Specified session id: {} has expired and is invalid.'.format(session_id))
+
+        super().prepare()
+
+    def write(self, response):
```

Comment:
That would also set the Session-Id on the slave services, and because the slave service is on another process, the Session-Id returned there would be different from the Session-Id generated on the master service.